doofinder-for-woocommerce/includes/class-landing.php

```diff
@@ -304,20 +293,17 @@ public function get_data_html($landing_slug)
     /**
      * Generates HTML content for displaying an error message on the landing page.
      *
-     * @param string $error The error message to display.
-     *
      * @return string The HTML content for the error message.
      */
-    public function get_error_html($error)
+    public function get_error_html()
     {
         ob_start();
     ?>
         <div id="primary" class="df-content-area content-area">
              <main id="main" class="site-main">
                 <div class="df-error-col content">
-                    <h3> We couldnt retrieve the API data. Please try again later. </h3>
                     <p>
-                        <?php echo $error; ?>
+                        <?php echo self::ERROR_MSG; ?>
```

Comment:
It would be helpful using translation functions here to print the error message, so in the future when we translate the plugin, we won't have to update the code.

```suggestion
                        <?php _e(self::ERROR_MSG, 'wordpress-doofinder'); ?>
```

Reply:
I agree with this idea, but I'm afraid that WordPress gettext functions don't allow dynamic translations, so instead of this change, I suggest another one:

`const ERROR_MSG = __("[500] - Oops! We're experiencing some technical difficulties at the moment. Please check back later. We apologize for any inconvenience.", "wordpress-doofinder");`

Reply:
I don't see why the given example wouldn't work, could you explain the problem @davidmolinacano ?

Reply:
Yes, of course :smile: I also faced this limitation in my previous company. Functions like __(), _e(), _n() expects the first parameter to be a fixed string instead of a variable. If a variable is used, several plugins (like Loco Translate) or other tools are unable to extract the translatable string. If only part of the text is variable, `sprintf` + %s should be used instead.

Example:

```php
<?php
$weekday = __('Monday', 'wordpress-doofinder');
$fulltext = sprintf(__('The week starts on %s', 'wordpress-doofinder'), $weekday);
```

Reply:
OK, you're right, it seems that the translation plugins search the code for the string literals in order to register them.
Thanks for the explanation.

Reply:
Anyway, using the `__()` function inside a const declaration will produce a fatal error, so, I think that we should repeat the string if we want to translate it. Otherwise, we could leave the code as is by the moment in order to avoid repeating the same string.
I leave the decision to you @eduardogomez97.

Reply:
Ah, you're totally right, I didn't think about that. Thank you for the correction @pedromcp90 :raised_hands:

Reply:
Thanks! @pedromcp90 @davidmolinacano